test/puppet/puppet_cache_test.rb

```diff
@@ -1,19 +1,21 @@
 require 'test_helper'
+require 'puppet_proxy/puppet_plugin'
 require 'puppet_proxy/puppet_class'
 require 'puppet_proxy/class_scanner_base'
 require 'puppet_proxy/puppet_cache'
+require 'tmpdir'
 
 class PuppetCacheTest < Test::Unit::TestCase
 
   def setup
     @scanner = Proxy::Puppet::ClassScannerBase.new
     @scanner.stubs(:scan_manifest).returns([Proxy::Puppet::PuppetClass.new('testinclude')]).then.
-                                   returns([Proxy::Puppet::PuppetClass.new('testinclude::sub::foo')])
-    Proxy::Puppet::PuppetCache.stubs(:write_to_cache)
+        returns([Proxy::Puppet::PuppetClass.new('testinclude::sub::foo')])
   end
 
   def test_should_refresh_cache_when_dir_is_not_in_cache
-    Proxy::Puppet::PuppetCache.stubs(:read_from_cache).returns({})
+    Proxy::Puppet::PuppetCache.expects(:read_from_cache).returns({})
+    Proxy::Puppet::PuppetCache.expects(:write_to_cache)
```

Comment:
Would you mind adding `.with` to this and the test below to ensure what's being written to the cache is sane?